app/models/plant_line.rb

```diff
@@ -30,10 +30,14 @@ def self.grid_data(filter)
       'data_owned_by',
       'organisation'
 
-    query = where(plant_line_name: filter[:plant_line_names]) if filter[:plant_line_names].present?
+    query = where(filter[:query]) if filter[:query].present?
     query = where('plant_line_name ILIKE ?', "%#{filter[:search]}%") if filter[:search].present?
     query ||= none
 
+    filter[:query].each do |k,_|
```

Comment:
Well, we indeed might need multiple-criteria testing (I made a spec for that) but, what is more important, I plan to refactor this to make that code work cross-model - for every model class that should be datatablesable ;). Having it generic helps.

For the time being I rely on strong parameters here - and I also added a spec for that (testing on PL case) - but I admit this might make it vulnerable for e.g. future API calls (if not outfitted with strong parameters). Let me consider  additional check after I abstract this code up, since the check will be model-specific. Setting another issue for that.